src/Microsoft.TestPlatform.Utilities/InferRunSettingsHelper.cs

```diff
@@ -377,6 +379,40 @@ private static List<string> GetNodeAttributes(XPathNavigator node)
             return null;
         }
 
+        public static Dictionary<string, string> GetEnvironmentVariables(string runSettings)
+        {
+            var environmentVariables = new Dictionary<string, string>();
+            try
+            {
+                using (var stream = new StringReader(runSettings))
+                using (var reader = XmlReader.Create(stream, XmlRunSettingsUtilities.ReaderSettings))
+                {
+                    var document = new XmlDocument();
+                    document.Load(reader);
+                    var runSettingsNavigator = document.CreateNavigator();
+
+                    var node = runSettingsNavigator.SelectSingleNode(EnvironmentVariablesNodePath);
+                    if (node == null)
+                    {
+                        return environmentVariables;
+                    }
+
+                    var childNodes = node.SelectChildren(XPathNodeType.Element);
+
+                    while (childNodes.MoveNext())
+                    {
+                        environmentVariables.Add(childNodes.Current.Name, childNodes.Current.Value);
```

Comment:
Checks

Reply:
This comes out to be always empty and never null.
The following is invalid  and document fails to load.
```
<RunConfiguration>
 <EnvironmentVariables>
   <Foo>
 </EnvironmentVariables>
</RunConfiguration>
```